PrestaShop/PrestaShop
# Fix base image format generation, enable it for new image system, simplify

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Read below
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | 
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/31488
| Related PRs       | 
| Sponsor company   |

### Description
- Multiple issues arose when implementing the image generation. This should fix everything.
- The new image system broke image generation with the old one. Regardless on what was set in `PS_IMAGE_QUALITY` it was always generating a JPG file and completely ignoring what was inside the file.
- It didn't work at all on legacy product page for example.
- Also, this PR enables to choose the base format for files with JPG extension when using the new generation system.
- I fixed the multiple images generation everywhere where it was previously implemented. If it was not implemented, I didn't implement it, because it would be generated in FO automatically.

### Settings with new image system
When enabling new image system, the settings are changed a bit. You can now configure the format for base PNG images and the new checkboxes.
![newimagesettings](https://user-images.githubusercontent.com/6097524/220628333-ea374e93-3b6c-421a-a2db-cbb75c9a2fb9.jpg)

### Summary
- Old system - we generate:
  - JPG file with JPG/PNG/WEBP image inside
- New system - we generate:
  - JPG file with JPG/PNG image inside
  - optionally WEBP file with WEBP inside
  - optionally AVIF file with AVIF inside

### How to test - GENERAL
- To easily test it, download https://www.xnview.com/en/xnconvert/ and just drop whole folders into there and you will immediately see the file formats inside.
- Test generation when uploading on old product page.
- Test generation when uploading on new product page.
- Test manual regeneration in backoffice.
- Test automatic generation in front office. Just delete thumbnails and visit the product, they should be regenerated automatically.
- Product images should automatically generate on upload, but others may not. 
- The main thing to check that it generates nothing **WRONG** in backoffice. If it generates only jpg **PROPERLY** (with correct format inside) and the rest on the fly in front office, it's OK.

### How to test - OLD IMAGE SYSTEM
- Disable new multiple image generation feature flag.
- Set image settings to `Use PNG only if the base image is in PNG format`.
- Make a product, upload one JPG and one PNG image.
- Go to FO to this product to generate the thumbnails.
- Go to img/p folder and find the folder with images. Drop them all into XNCONVERT or simmilar software, or anything that will let you see what is inside. See that the thumbnail contain proper files.
- Then you can try again with "Use PNG" or "Use JPG" options. Just delete all the thumbnails and refresh the product in FO, so you don't have to create it again.

### How to test - NEW IMAGE SYSTEM
- Enable new multiple image generation feature flag.
- Set image settings to `Use PNG only if the base image is in PNG format` and enable webp AND/OR AVIF.
- Make a product, upload one JPG and one PNG image.
- Go to FO to this product to generate the thumbnails.
- Go to img/p folder and find the folder with images. Drop them all into XNCONVERT or simmilar software, or anything that will let you see what is inside. See that the thumbnail contain proper files. It will look something like this:
![filetypes](https://user-images.githubusercontent.com/6097524/220628128-63934ea4-3d23-49e5-8693-e9a7efa723fe.jpg)

### Known issues
- If you are on old image system and set image format to "WEBP" or "webp when original image is webp", then go enable new image system, when you go to image settings, you may not see the selected value. Everything will work, but you will get two WEBP images, one with jpg extension, one with webp extension. This can get fixed in another PR, I didn't want to mess with the experimental feature enabling.
